### PR TITLE
amm: constructor checks tests

### PR DESCRIPTION
**Pull request by mijovic**

### CODE REVIEW

File: contracts/amm/TempusAMM.sol [hunk 1]
```diff
@@ -77,6 +77,7 @@ contract TempusAMM is ITempusAMM, ERC20, Pausable, Ownable {
     ) ERC20(name, symbol) {
         require(amplificationStartValue >= MIN_AMPLIFICATION, "min amp");
         require(amplificationStartValue <= MAX_AMPLIFICATION, "max amp");
+        require(swapFeePerc < 1e18, "max swap fee");
```
Comment (yuval-weiss) [inline contracts/amm/TempusAMM.sol:80]: better to limit swap fee to a lower number (maybe 5%?) imo

Comment (yuval-weiss) [reply contracts/amm/TempusAMM.sol:80]: and also -- let's add a constant variable for that

**Comment by mijovic** [comment]
@yuval-weiss updated

File: test/amm/TempusAMM.test.ts [hunk 2]
```diff
@@ -86,6 +88,55 @@ describeForEachPool("TempusAMM", (testFixture:PoolTestFixture) =>
...
+      ammAmplifyStart: 0.5,
+      ammAmplifyEnd: 10
+    }))).to.equal("min amp");
+
+    (await expectRevert(testFixture.createWithAMM({
+      initialRate:1.0, poolDuration:ONE_MONTH, yieldEst:0.1,
+      ammSwapFee:SWAP_FEE_PERC,
+      ammAmplifyStart: 5,
+      ammAmplifyEnd: 1000000
+    }))).to.equal("max amp");
```
[the remark this answers is not in the document]
Comment (mijovic) [reply test/amm/TempusAMM.test.ts:104]: will add a check for end value as well

[the remark this answers is not in the document]
Comment (yuval-weiss) [reply test/amm/TempusAMM.test.ts:104]: weird. Other than this thing, this PR looks good to me

[the remark this answers is not in the document]
Comment (mijovic) [reply test/amm/TempusAMM.test.ts:111]: Pushed additional test that fails here

File: contracts/amm/TempusAMM.sol [hunk 3]
```diff
@@ -43,6 +43,9 @@ contract TempusAMM is ITempusAMM, ERC20, Pausable, Ownable {
     uint256 private constant MIN_AMPLIFICATION = 1000;
     uint256 private constant MAX_AMPLIFICATION = 5000000;
 
+    // Maximum swap fee is set to 5%
+    uint256 private constant MAX_SWAP_FEE = 5e16;
```
Comment (axic) [inline contracts/amm/TempusAMM.sol:47]:
```suggestion
    uint256 private constant MAX_SWAP_FEE_PERCENTAGE = 5e16;
```

but maybe even this works:
```suggestion
    uint256 private constant MAX_SWAP_FEE_PERCENTAGE = 0.05e18;
```

File: test/amm/TempusAMM.test.ts [hunk 4]
```diff
@@ -86,6 +88,62 @@ describeForEachPool("TempusAMM", (testFixture:PoolTestFixture) =>
...
+      initialRate:1.0, poolDuration:ONE_MONTH, yieldEst:0.1,
+      ammSwapFee:SWAP_FEE_PERC,
+      ammAmplifyStart: 0.5,
+      ammAmplifyEnd: 10
+    }))).to.equal("min amp");
+
+    (await expectRevert(testFixture.createWithAMM({
+      initialRate:1.0, poolDuration:ONE_MONTH, yieldEst:0.1,
+      ammSwapFee:SWAP_FEE_PERC,
+      ammAmplifyStart: 1000000,
```
Comment (axic) [inline test/amm/TempusAMM.test.ts:102]: This is below max, why would this fail?

Comment (mijovic) [reply test/amm/TempusAMM.test.ts:102]: this is raw, it is then multiplied by `AMP_PRECISION` in test suite

**Review by axic — APPROVED** [review]

**Pull request merged**